Lib/multiprocessing/popen_fork.py

```diff
@@ -66,17 +66,22 @@ def kill(self):
     def _launch(self, process_obj):
         code = 1
         parent_r, child_w = os.pipe()
+        child_r, parent_w = os.pipe()
         self.pid = os.fork()
         if self.pid == 0:
             try:
                 os.close(parent_r)
-                code = process_obj._bootstrap()
+                os.close(parent_w)
+                code = process_obj._bootstrap(parent_sentinel=child_r)
             finally:
                 os._exit(code)
         else:
             os.close(child_w)
+            os.close(child_r)
             self.finalizer = util.Finalize(self, os.close, (parent_r,))
+            self.finalizer = util.Finalize(self, os.close, (parent_w,))
```

Comment:
Hmm, this is clobbering the previous finalizer. You should use a single callback that closes both fds.